language/diem-framework/modules/NetworkIdentity.move

```diff
@@ -140,25 +140,177 @@ module DiemFramework::NetworkIdentity {
     }
     spec remove_identities {
         let account_addr = Signer::spec_address_of(account);
-        let num_identities = len(global<NetworkIdentity>(account_addr).identities);
-        // aborts_if len(to_remove) == 0;
-        // aborts_if len(to_remove) > MAX_ADDR_IDENTITIES;
+        let prior_identities = global<NetworkIdentity>(account_addr).identities;
+        let has_change = (exists i in 0..len(to_remove): contains(prior_identities, to_remove[i]));
+
+        let post handle = global<NetworkIdentity>(account_addr).identity_change_events;
+        let post msg = NetworkIdentityChangeNotification {
+            identities: global<NetworkIdentity>(account_addr).identities,
+            time_rotated_seconds: DiemTimestamp::spec_now_seconds(),
+        };
+
+        aborts_if len(to_remove) == 0;
+        aborts_if len(to_remove) > MAX_ADDR_IDENTITIES;
+        aborts_if !exists<NetworkIdentity>(account_addr);
+        include has_change ==> DiemTimestamp::AbortsIfNotOperating;
+        include RemoveMembersInternalEnsures<vector<u8>> {
+            old_members: prior_identities,
+            new_members: global<NetworkIdentity>(account_addr).identities,
+        };
         modifies global<NetworkIdentity>(account_addr);
-        invariant exists<NetworkIdentity>(account_addr);
-        invariant num_identities <= MAX_ADDR_IDENTITIES;
+        emits msg to handle if has_change;
+    }
+
+    // =================================================================
```

Comment:
I'm wondering whether we should move this to the Vector module for reuse as long as we do not have sets? Can also be done in a separate PR.

@sblackshear @tzakian